k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -137,13 +137,77 @@ pub struct GetUpgradeArgs {}
 impl GetUpgradeArgs {
     ///  Upgrade the resources.
     pub async fn get_upgrade(&self, namespace: &str) {
-        println!("{namespace}")
-        // To be implemented
+        // Create resources for getting upgrade status
+        match GetUpgradeEventClient::create_get_upgrade_resource(namespace).await {
+            Ok(_) => {}
+            Err(error) => eprintln! {"Error in getting upgrade status: {error}"},
+        }
+    }
+}
+
+/// Resource to be created to get upgrade status.
+struct GetUpgradeEventClient {
+    upgrade_event: Api<Event>,
```

Comment:
`upgrade_event` ---> `api`. Why its is called `GetUpgradeEventClient` ? Why not just call it `UpgradeEventClient`

Reply:
Adding the `Get` keyword binds it to only get operation, which make it not extendable in future.